cirq-core/cirq/sim/density_matrix_simulator.py

```diff
@@ -119,7 +123,7 @@ def __init__(
         noise: 'cirq.NOISE_MODEL_LIKE' = None,
         seed: 'cirq.RANDOM_STATE_OR_SEED_LIKE' = None,
         ignore_measurement_results: bool = False,
-        split_untangled_states: bool = False,
+        split_untangled_states: bool = True,
```

Comment:
Sparse simulator was changed to True by default as well. The reason we had it False was because the `StepResult.set_state_vector` broke if it was True. But this PR fixes the set_state_vector issue so it's fully backwards compatible.